# Logistic Regression Example

This notebook demonstrates a simple Logistic Regression model using sklearn.
We will use the breast cancer dataset and evaluate the model with accuracy and confusion matrix.


In [3]:
import numpy as np
import pandas as pd
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix


## Dataset

We use the Breast Cancer Wisconsin dataset provided by scikit-learn.
It contains numerical features computed from digitized images of breast masses.

In [4]:

data = load_breast_cancer()
X = data.data
y = data.target


X.shape, y.shape


((569, 30), (569,))

The dataset contains 569 samples with 30 features.
The target variable has two classes: malignant and benign.


In [6]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


We split the data into 80% training and 20% test sets.

In [8]:
model = LogisticRegression(max_iter=10000)
model.fit(X_train, y_train)


LogisticRegression(max_iter=10000)

We train a Logistic Regression model. max_iter=10000 ensures convergence.

In [10]:
y_pred = model.predict(X_test)

acc = accuracy_score(y_test, y_pred)
cm = confusion_matrix(y_test, y_pred)

print("Accuracy:", acc)
print("Confusion Matrix:\n", cm)


Accuracy: 0.956140350877193
Confusion Matrix:
 [[39  4]
 [ 1 70]]


The accuracy shows the overall performance of the model.
The confusion matrix provides a more detailed view of correct and incorrect predictions for each class.
